bin/ycsb.sh

```diff
@@ -0,0 +1,183 @@
+  exit 1;
+fi
+
+# Find binding information
+BINDING_LINE=`grep "^$2:" "$YCSB_HOME/bin/bindings.properties" -m 1`
+
+if [ -z $BINDING_LINE ] ; then
+  echo "[ERROR] The specified binding '$2' was not found.  Exiting."
+  exit 1;
+fi
+
+# Get binding name and class
+BINDING_NAME=`echo $BINDING_LINE | cut -d':' -f1`
+BINDING_CLASS=`echo $BINDING_LINE | cut -d':' -f2`
+
+# Some bindings have multiple versions that are managed in the same directory.
+#   They are noted with a '-' after the binding name.
+#   (e.g. cassandra-7 & cassandra-8)
+BINDING_DIR=`echo $BINDING_NAME | cut -d'-' -f1`
+
+# The 'basic' binding is core functionality
+if [ $BINDING_NAME = "basic" ] ; then
+  BINDING_DIR=core
+fi
+
+# For Cygwin, ensure paths are in UNIX format before anything is touched
+if $CYGWIN; then
+  [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
+  [ -n "$CLASSPATH" ] && CLASSPATH=`cygpath --path --unix "$CLASSPATH"`
+fi
+
+# Check if source checkout, or release distribution
+DISTRIBUTION=true
+if [ -r "$YCSB_HOME/pom.xml" ]; then
+  DISTRIBUTION=false;
+fi
+
+# Add Top level conf to classpath
+if [ -z "$CLASSPATH" ] ; then
+  CLASSPATH="$YCSB_HOME/conf"
+else
+  CLASSPATH="$CLASSPATH:$YCSB_HOME/conf"
+fi
+
+# Build classpath
+if $DISTRIBUTION; then
+  # Core libraries
+  for f in `ls "$YCSB_HOME"/lib/*.jar`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database conf dir
+  if [ -r "$YCSB_HOME/$BINDING_DIR-binding/conf" ] ; then
+    CLASSPATH="$CLASSPATH:$YCSB_HOME/$BINDING_DIR-binding/conf"
+  fi
+
+  # Database libraries
+  for f in `ls "$YCSB_HOME"/$BINDING_DIR-binding/lib/*.jar 2>/dev/null`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+else
+  # Core libraries
+  for f in `ls "$YCSB_HOME"/core/target/*.jar`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database conf (need to find because location is not consistent)
+  for f in `find "$YCSB_HOME"/$BINDING_DIR -name "conf"`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database libraries
+  for f in `ls "$YCSB_HOME"/$BINDING_DIR/target/*.jar`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database dependency libraries
+  for f in `ls "$YCSB_HOME"/$BINDING_DIR/target/dependency/*.jar 2>/dev/null`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+fi
+
+# Cassandra deprecation message
+if [ $BINDING_DIR = "cassandra" ] ; then
+  echo "[WARN] The 'cassandra-7', 'cassandra-8', 'cassandra-10', and \
+cassandra-cql' clients are deprecated. If you are using \
+Cassandra 2.X try using the 'cassandra2-cql' client instead."
+fi
+
+# For Cygwin, switch paths to Windows format before running java
+if $CYGWIN; then
+  JAVA_HOME=`cygpath --absolute --windows "$JAVA_HOME"`
+  CLASSPATH=`cygpath --path --windows "$CLASSPATH"`
+fi
+
+# Get the rest of the arguments
+YCSB_ARGS=`echo $* | cut -d' ' -f3-`
+
+# Run YCSB
+echo \"$JAVA_HOME/bin/java\" $JAVA_OPTS -classpath \"$CLASSPATH\" $YCSB_CLASS $YCSB_COMMAND -db $BINDING_CLASS $YCSB_ARGS
```

Comment:
before this line, we should confirm that JAVA_HOME works, and that CLASSPATH is non-empty.